### ISSUE TITLE
findJSFile  matching logic is liable to break

### ISSUE THREAD

**Issue by padraicbc** [plentico/plenti#144]
I had added a break statement [after](https://github.com/plentico/plenti/blob/master/cmd/build/gopack.go#L200)  as I presumed the first match was the only match. I ended up gettng `Uncaught SyntaxError: The requested module '/spa/web_modules/svelte/compiler.js' does not provide an export named 'onMount'` when the file I really wanted was `public/spa/web_modules/svelte/index.js`. The winner is it is determined by a lexicographical sort, highest wins so say along with `regisiter.js` there was a `register.mjs` in the svelte directory. We would have the follwoing output:

```
The found import path to use is: public/spa/web_modules/svelte/compiler.js
The found import path to use is: public/spa/web_modules/svelte/index.js
The found import path to use is: public/spa/web_modules/svelte/register.js
```

Which would have the wrong file again and now give `header.js:36 Uncaught SyntaxError: The requested module '/spa/web_modules/svelte/register.js' does not provide an export named 'onMount'`.  Whatever matching file that comes last when sorted by filename will essentially be the file we use. I am guessing this can break easily?

**Comment by jimafisk** [comment]
You're 100% right about this, it's only working because of the file structure of the current version of svelte. At one point we were having issues importing from the base `svelte` package (see https://github.com/plentico/plenti/issues/38 and https://github.com/plentico/plenti/issues/92), so I made the change where we check the all the files in the current directory for a ".js" or ".mjs" file before moving up to a nested directory and performing the same search. That doesn't guarantee it would find an index.js vs a compiler.js or register.js like you said. The reason it doesn't match register.js is Svelte does not ship with a `.mjs` version of that file so gopack never copies it over to `public/spa/web_modules`.

If I implement the onMount tutorial (https://svelte.dev/tutorial/onmount) in Plenti (just make sure to replace backtic characters with quotes because of this bug: https://github.com/plentico/plenti/issues/121), it does "work":

<details>
<summary>Website screenshot</summary>

![photos](https://user-images.githubusercontent.com/5913244/112666353-778a6680-8e32-11eb-831c-28330e17dffa.png)

</details>

<details>
<summary>Import screenshot</summary>

![onmount](https://user-images.githubusercontent.com/5913244/112666418-8b35cd00-8e32-11eb-8859-8645c4e08f2f.png)

</details>

But we're basically relying on the good grace of the sort order like you mentioned above. This is fragile and should be improved, good catch!

**Comment by jimafisk** [comment]
This should be fixed in `v0.4.27` (successful release [v0.4.29](https://github.com/plentico/plenti/releases/tag/v0.4.29)). Gopack is now actually reading the `package.json` file of npm modules and looking for the `module` key to determine the esm entrypoint:

https://github.com/plentico/plenti/blob/0a14d97a01182f9a3f5948a872e58196c7cb862c/cmd/build/gopack.go#L208

In the future we could make this more robust to account for the different ways you can specify esm support in `package.json` discussed here: https://github.com/plentico/plenti/issues/165#issuecomment-974998215. I've even thought about releasing Gopack as a separate tool, in which case we may want to incorporate esbuild's [format](https://esbuild.github.io/api/#format) from their [transform-api](https://esbuild.github.io/api/#transform-api) to convert commonjs into esm if the module maintainer didn't did this themselves: https://esbuild.github.io/api/#format-esm
